src/Spatial/Euclidean/Sphere3D.cs

```diff
@@ -0,0 +1,245 @@
+        /// </summary>
+        public readonly double Radius;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Sphere3D"/>.
+        /// </summary>
+        /// <param name="centerPoint">The center of the Sphere</param>
+        /// <param name="radius">The radius of the Sphere</param>
+        public Sphere3D(Point3D centerPoint, double radius)
+        {
+            // Anypoint is same to other point.
+            if (radius <= 0)
+            {
+                throw new ArgumentException("The radius is negative.");
+            }
+
+            this.CenterPoint = centerPoint;
+            this.Radius = radius;
+        }
+
+        /// <summary>
+        /// Gets the diameter of the sphere.
+        /// </summary>
+        [Pure]
+        public double Diameter => 2 * Radius;
+
+        /// <summary>
+        /// Gets the circumference of the sphere.
+        /// </summary>
+        [Pure]
+        public double Circumference => 2 * Math.PI * Radius;
+
+        /// <summary>
+        /// Gets the volume of the sphere.
+        /// </summary>
+        [Pure]
+        public double Volume => 4 / 3 * Math.PI * this.Radius * this.Radius * this.Radius;
+
+        /// <summary>
+        /// Gets the area of the sphere.
+        /// </summary>
+        [Pure]
+        public double Area => 4 * Math.PI * this.Radius * this.Radius;
+
+        /// <summary>
+        /// Returns a value that indicates whether each pair of elements in two specified sphere is equal.
+        /// </summary>
+        /// <param name="left">The first sphere to compare</param>
+        /// <param name="right">The second sphere to compare</param>
+        /// <returns>True if the sphere are the same; otherwise false.</returns>
+        public static bool operator ==(Sphere3D left, Sphere3D right)
+        {
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// Returns a value that indicates whether any pair of elements in two specified sphere is not equal.
+        /// </summary>
+        /// <param name="left">The first sphere to compare</param>
+        /// <param name="right">The second sphere to compare</param>
+        /// <returns>True if the sphere are different; otherwise false.</returns>
+        public static bool operator !=(Sphere3D left, Sphere3D right)
+        {
+            return !left.Equals(right);
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="Sphere3D"/>sphere which lie along its circumference of tetrahydron made of four points.
+        /// </summary>
+        /// <param name="p1">The first point on the sphere</param>
+        /// <param name="p2">The second point on the sphere</param>
+        /// <param name="p3">The third point on the sphere</param>
+        /// <param name="p4">The last point on the sphere</param>
+        public static Sphere3D FromFourPoints(Point3D p1, Point3D p2, Point3D p3, Point3D p4)
+        {
```

Comment:
I would favor an implementation which is easier to follow, based for example on intersections of planes and lines